datasets/food101/food101.py

```diff
@@ -0,0 +1,192 @@
+    "foie_gras",
+    "french_fries",
+    "french_onion_soup",
+    "french_toast",
+    "fried_calamari",
+    "fried_rice",
+    "frozen_yogurt",
+    "garlic_bread",
+    "gnocchi",
+    "greek_salad",
+    "grilled_cheese_sandwich",
+    "grilled_salmon",
+    "guacamole",
+    "gyoza",
+    "hamburger",
+    "hot_and_sour_soup",
+    "hot_dog",
+    "huevos_rancheros",
+    "hummus",
+    "ice_cream",
+    "lasagna",
+    "lobster_bisque",
+    "lobster_roll_sandwich",
+    "macaroni_and_cheese",
+    "macarons",
+    "miso_soup",
+    "mussels",
+    "nachos",
+    "omelette",
+    "onion_rings",
+    "oysters",
+    "pad_thai",
+    "paella",
+    "pancakes",
+    "panna_cotta",
+    "peking_duck",
+    "pho",
+    "pizza",
+    "pork_chop",
+    "poutine",
+    "prime_rib",
+    "pulled_pork_sandwich",
+    "ramen",
+    "ravioli",
+    "red_velvet_cake",
+    "risotto",
+    "samosa",
+    "sashimi",
+    "scallops",
+    "seaweed_salad",
+    "shrimp_and_grits",
+    "spaghetti_bolognese",
+    "spaghetti_carbonara",
+    "spring_rolls",
+    "steak",
+    "strawberry_shortcake",
+    "sushi",
+    "tacos",
+    "takoyaki",
+    "tiramisu",
+    "tuna_tartare",
+    "waffles",
+]
+
+
+class Food101(datasets.GeneratorBasedBuilder):
+    """Food-101 Images dataset."""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "image": datasets.Value("string"),
+                    "label": datasets.features.ClassLabel(names=_NAMES),
+                }
+            ),
+            supervised_keys=("image", "label"),
+            homepage=_HOMEPAGE,
+            task_templates=[ImageClassification(image_file_path_column="image", label_column="label", labels=_NAMES)],
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        dl_path = Path(dl_manager.download_and_extract(_BASE_URL))
+        meta_path = dl_path / "food-101" / "meta"
+        image_dir_path = dl_path / "food-101" / "images"
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"json_file_path": meta_path / "train.json", "image_dir_path": image_dir_path},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"json_file_path": meta_path / "test.json", "image_dir_path": image_dir_path},
+            ),
+        ]
+
+    def _generate_examples(self, json_file_path, image_dir_path):
+        """Generate images and labels for splits."""
+        labels = self.info.features["label"]
+        data = json.loads(json_file_path.read_text())
+        for label, images in data.items():
+            for image_name in images:
+                image = image_dir_path / f"{image_name}.jpg"
+                features = {"image": str(image), "label": labels.encode_example(label)}
```

Comment:
You don't need to call `Features.encode_example` because that is done under the hood by the builder.

Reply:
Great! I'll get rid of it